src/backend/distributed/executor/repartition_join_execution.c

```diff
@@ -71,6 +75,20 @@ ExecuteDependentTasks(List *topLevelTasks, Job *topLevelJob)
 }
 
 
+/*
+ * EnsureNoVisibilityProblem makes sure that the tasks won't have
+ * any visibility problems.
+ */
+static void
+EnsureNoVisibilityProblem(List *taskList)
+{
+	if (TransactionAccessedLocalPlacement && AnyTaskAccessesLocalNode(taskList))
+	{
+		ErrorIfTransactionAccessedPlacementsLocally();
```

Comment:
We should probably rename this function. It looks a lot like `TransactionAccessedLocalPlacement`, which is confusing.